MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,54 @@
+# Migration Guide
+Changes are grouped as follows:
```

Comment:
^First draft of migration guide added. After format is decided on I will finalize the text.

Reply:
Note to self: Add note on all parameters now being keyword-only (also in changelog)